File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Deque<Character> stack = new ArrayDeque<>();
        for(int i=0;i<s.length();i++){
            char c = s.charAt(i); 
            if(c == '(' || c == '{' || c == '['){
                stack.push(c);
            }
            else{
                if(stack.isEmpty() == true){
                    return false;
                }
                else if(matching(stack.peek(), c) == false){
                    return false;
                }
                else{
                    stack.pop();
                }
            }
        }
        
        return stack.isEmpty();
    }
    
    static boolean matching(char a, char b){
        return ((a == '(' && b == ')') || (a == '{' && b == '}') || (a == '[' && b == ']'));
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    for(let i=0;i<s.length;i++){
        if(s.charAt(i) == '{' || s.charAt(i) == '(' || s.charAt(i) == '[' ){
            stack.push(s.charAt(i));
        }
        else{
            if(stack.length == 0){
                return false;
            }
            else if(matching(stack[stack.length-1], s.charAt(i)) == false)
                return false;
            else
                stack.pop();
        }
    }
        return stack.length == 0;
};

let matching = function(a,b){
    return ((a == '{' && b == '}') ||(a == '(' && b == ')') || (a == '[' && b == ']'));
}